include/drivers/clock_control/nrf_clock_control.h

```diff
@@ -10,7 +10,21 @@
 #include <device.h>
 #include <hal/nrf_clock.h>
 
-/* TODO: move all these to clock_control.h ? */
+/** @brief Clocks handled by the CLOCK peripheral.
+ *
+ * Enum shall be used as a sys argument in clock_control API.
+ */
+enum nrf_clock_type {
+	NRF_CLOCK_TYPE_HFCLK,
```

Comment:
I think we should keep prefixes `nrf_clock_` and `NRF_CLOCK_` reserved for the stuff defined in `<hal/nrf_clock.h>`. `CLOCK_CONTROL_NRF_` seems to suit here.

Reply:
sure, i should respect nrfx copyrights ;)